src/averageAges.js

```diff
@@ -15,17 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menDiedInCentury = people.filter(person => person.sex === 'm'
+  && Math.ceil(person.died / 100) === century);
+
+  const allMen = people.filter(person => person.sex === 'm');
+
+  const men = (typeof (century) === 'undefined') ? allMen : menDiedInCentury;
```

Comment:
typeof is redundant here
```suggestion
  const men = century ? menDiedInCentury : allMen;
```